relayer/relayer.go

```diff
@@ -136,9 +136,7 @@ func NewRelayer(
 	}
 
 	if shouldProcessMissedBlocks {
-		// If StartBlockHeight is not set in the config, then we process from height 0
-		startHeight := big.NewInt(0).SetUint64(sourceSubnetInfo.StartBlockHeight)
-		err = r.processMissedBlocks(sub, startHeight)
+		err = r.processMissedBlocks(sub, sourceSubnetInfo.StartBlockHeightBigInt)
```

Comment:
I'm wondering if we could refactor this part of the code a bit. The issue is that I want to be able to process missed blocks after reconnecting to the subscriber after a subscriber error. The current flow doesn't really work for this. 

Could we create a function `setProcessedBlockHeight()` that takes `shouldProcessMissedBlocks` `sourceSubnetInfo.StartBlockHeightBigInt`, and then does all the logic of checking/setting `latestProcessedBlock` in the db that `processMissedBlocks`, or the call to `SetProcessedBlockHeightToLatest` below are currently doing? 

We could then just call `sub.ProcessFromHeight(<whatever is in the db>)` in every case, and we could also call it after reconnection.

Reply:
That's an interesting idea. Slight alteration - I'd replace `sub.ProcessFromHeight(*big.Int)` with `sub.ProcessFromStoredBlock()` so the usage is:
1. Set the starting block in the DB by choosing between the existing value and the config value, according to the established logic
2. call `sub.ProcessFromStoredBlock()`

I'll put this together, and see how it looks.

Reply:
Yes that sounds great! Thank you!

Reply:
Revisiting this code, I don't see any reason why the subscriber needs to access the database at all. This was added before marking a block as processed was outsourced from the subscriber to the relayer.

In this case, your original suggestion makes the most sense. The subscriber will only execute `eth_getLog` queries to add blocks to the process queue, but the relayer will be the entity that processes them and marks them as processed in the db.

Lmk what you think.

Reply:
Yes, that sounds great!